### PR TITLE
Clear issue status on label removal from user

### PR DESCRIPTION
**Pull request by hubertdeng123**
When a user removes a `Waiting for: *` label from an issue, clear status in `Issues Someone Else Cares About` board. If it's a bot, don't clear, since the bot will only unlabel issues when it is about to add another one.

### CODE REVIEW

File: test/github-orgs.yml [hunk 1]
```diff
@@ -4,11 +4,11 @@ getsentry:
     privateKey: 'STUB'
     installationId: -1
   project:
-    nodeId: ''
```
Comment (hubertdeng123) [inline test/github-orgs.yml:7]: makes assertions more accurate in tests to populate these with values

File: src/brain/issueLabelHandler/index.test.ts [hunk 2]
```diff
@@ -748,5 +768,37 @@ describe('issueLabelHandler', function () {
...
+    it.each([
+      WAITING_FOR_PRODUCT_OWNER_LABEL,
+      WAITING_FOR_SUPPORT_LABEL,
+      WAITING_FOR_COMMUNITY_LABEL,
+    ])(
+      "should clear project issue status field if user removes '%s'",
+      async function (label) {
+        await setupIssue();
+        await addLabel(label, 'routing-repo');
+        await removeLabel(label);
```
Comment (azaslavsky) [inline src/brain/issueLabelHandler/index.test.ts:781]: nit: we only use this in two places, here and the test below. But the actual object its building to put in `makePayload` is actually pretty dramatically different - enough that you have 3 args for a function that only gets used twice. Imo its easier to read, and makes it clear what the test is doing.

Classic memo on not trying to make tests to DRY seems pretty applicable here: https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html

File: src/brain/issueLabelHandler/followups.ts [hunk 3]
```diff
@@ -105,6 +105,49 @@ export async function updateCommunityFollowups({
   tx.finish();
 }
 
+export async function clearWaitingForProductOwnerStatus({
+  id,
+  payload,
+  ...rest
+}: EmitterWebhookEvent<'issues.unlabeled'>) {
+  const tx = Sentry.startTransaction({
```
Comment (azaslavsky) [inline src/brain/issueLabelHandler/followups.ts:113]: Eating our own 🐶 🥫 , nice!

**Review by azaslavsky — APPROVED** [review]
Approved, with minor comments.

File: src/brain/issueLabelHandler/index.test.ts (hunk 2 above)
Comment (hubertdeng123) [reply src/brain/issueLabelHandler/index.test.ts]: fair enough, inlined

File: src/brain/issueLabelHandler/followups.ts [hunk 4]
```diff
@@ -105,6 +105,49 @@ export async function updateCommunityFollowups({
...
+  if (await shouldSkip(payload, org, reasonsToDoNothing)) {
+    return;
+  }
+
+  const { label } = payload;
+  const repo = payload.repository.name;
+  const issueNumber = payload.issue.number;
+  // Here label will never be undefined, ts is erroring here but is handled in the shouldSkip above
+  // @ts-ignore
+  const labelName = label.name;
```
[the remark this answers is not in the document]
Comment (hubertdeng123) [reply src/brain/issueLabelHandler/followups.ts:138]: fair point, changed

**Pull request merged**